src/webdriver.rs

```diff
@@ -188,3 +188,65 @@ where
         }
     }
 }
+
+#[cfg(feature = "persist")]
+pub mod persist {
+    use std::marker::PhantomData;
+
+    use serde::{Deserialize, Serialize};
+
+    use crate::{
+        common::types::SessionId,
+        error::WebDriverResult,
+        http::connection_async::WebDriverHttpClientAsync,
+        session::{spawn_session_task, WebDriverSession},
+        webdriver::{GenericWebDriver, WebDriver},
+    };
+
+    /// The persisted session that can be serialized and deserialized.
+    ///
+    /// # TODO
+    ///
+    ///  - What is `WebDriverConfig` used for? Do we need to persist it as well?
```

Comment:
The `WebDriverConfig` is for allowing things like `thirtyfour_query` where you can extend the functionality of the WebDriver through other traits, including storing configuration information (usually some default values) inside the `WebDriverSession`.

Right now it doesn't need to be persisted as it relates to how the session will be used rather than anything intrinsic to the session itself, but there is potential for that to change in future.

Personally I don't mind whether the config is persisted or not, but it is just a hashmap so easily serializable and deserializable, and likely to be very low-cost in terms of any extra disk usage. Maybe it's best just to persist it and then we don't need to worry about how people might use it. What are your thoughts?

Reply:
Okay, I just didn't know where it was used. Let's persist it then.